### PR TITLE
Create section 23 "Workflows" of the Controller UG

### PR DESCRIPTION
**Pull request by EMcWhinn**
Adding assembly and modules

Create section 23 of the Controller user Guide

https://issues.redhat.com/browse/AAP-15838

Affects `titles/controller-user-guide`

### CODE REVIEW

File: downstream/assemblies/platform/assembly-ug-controller-workflows.adoc [hunk 1]
```diff
@@ -0,0 +1,26 @@
+[id="controller-workflows"]
+
+= Workflows in {ControllerName}
+
+Workflows allow you to configure a sequence of disparate job templates (or workflow templates) that may or may not share inventory, playbooks, or permissions.
```
Comment (ianf77) [inline downstream/assemblies/platform/assembly-ug-controller-workflows.adoc:5]: enable you

File: downstream/modules/platform/con-controller-role-based-access-controls.adoc [hunk 2]
```diff
@@ -0,0 +1,13 @@
+[id="controller-role-based-access-controls"]
+
+= Role-based access controls
+
+To edit and delete a workflow job template, you must have the administrator role. 
+To create a workflow job template, you must be an organization administrator or a system administrator. 
+However, you can run a workflow job template that contains job templates that you do not have permissions for. 
+Organization administrators can create a blank workflow and then grant an `admin_role` to a low-level user, after which they can delegate more access and build the graph.
```
Comment (ianf77) [inline downstream/modules/platform/con-controller-role-based-access-controls.adoc:8]: Are organization administrators different to system administrators?

File: downstream/modules/platform/con-controller-role-based-access-controls.adoc [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+
+= Role-based access controls
+
+To edit and delete a workflow job template, you must have the administrator role. 
+To create a workflow job template, you must be an organization administrator or a system administrator. 
+However, you can run a workflow job template that contains job templates that you do not have permissions for. 
+Organization administrators can create a blank workflow and then grant an `admin_role` to a low-level user, after which they can delegate more access and build the graph. 
+You must have `execute` access to a job template to add it to a workflow job template.
+
+Other tasks such as the ability to make a duplicate copy and re-launch a workflow can also be performed, depending on what kinds of permissions are granted to a particular user. You must have permissions to all the resources used in a workflow, such as job templates, before relaunching or making a copy.
```
Comment (ianf77) [inline downstream/modules/platform/con-controller-role-based-access-controls.adoc:11]: You can also perform other tasks, such as making a duplicate copy or re-launching a workflow, depending on the which permissions are granted to a user. ?

File: downstream/modules/platform/con-controller-workflow-scenarios.adoc [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+For more information see the xref:controller-job-slicing[Job slicing] section.
+====
+
+* You can build a recursive workflow, but if {ControllerName} detects an error, it stops at the time the nested workflow attempts to run.
+* Artifacts gathered in jobs in the sub-workflow are passed to downstream nodes.
+* An inventory can be set at the workflow level, or prompt for inventory on launch.
+* When launched, all job templates in the workflow that have `ask_inventory_on_launch=true` use the workflow level inventory.
+* Job templates that do not prompt for inventory ignore the workflow inventory and run against their own inventory.
+* If a workflow prompts for inventory, schedules and other workflow nodes can provide the inventory.
+* In a workflow convergence scenario, `set_stats` data will be merged in an undefined way, so it is recommended that you set unique keys.
```
Comment (ianf77) [inline downstream/modules/platform/con-controller-workflow-scenarios.adoc:52]: set_stats data is merged in an undefined way, therefore you must set unique keys.    (Or what?)

File: downstream/modules/platform/con-controller-workflow-states.adoc [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+= Workflow states
+
+The workflow job can have the following states (no Failed state):
+
+* Waiting
+* Running
+* Success (finished)
+* Cancel
+* Error
+* Failed
```
Comment (ianf77) [inline downstream/modules/platform/con-controller-workflow-states.adoc:12]: Why is Failed here if there's no Failed state?

**Review by ianf77 — COMMENTED** [review]
Some comments

File: downstream/modules/platform/con-controller-workflow-states.adoc (hunk 5 above)
Comment (EMcWhinn) [reply downstream/modules/platform/con-controller-workflow-states.adoc:12]: I thought this was odd too, I need to ask an SME about this.

**Review by ianf77 — APPROVED** [review]
LGTM

**Pull request merged**